foundryup/foundryup

```diff
@@ -533,6 +557,73 @@ check_cmd() {
   command -v "$1" &>/dev/null
 }
 
+check_installer_up_to_date() {
+  say "checking if foundryup is up to date..."
+
+  tmp_file="$(mktemp)"
+  ensure download "$FOUNDRY_BIN_URL" "$tmp_file"
```

Comment:
that's fine but it actually downloads the foundryup in tem file, I think we could just have the `extract_installer_version` fn smth like

```bash
    if check_cmd curl; then
      curl -s https://raw.githubusercontent.com/foundry-rs/foundry/master/foundryup/foundryup | grep 'FOUNDRYUP_INSTALLER_VERSION="' | cut -d'"' -f2
    else
      wget -qO- https://raw.githubusercontent.com/foundry-rs/foundry/master/foundryup/foundryup | grep 'FOUNDRYUP_INSTALLER_VERSION="' | cut -d'"' -f2
    fi
```

Reply:
Good idea :+1: 

Updated in https://github.com/foundry-rs/foundry/pull/10952/commits/5ee447eaddd6e76ec3c4349d20eff79fd2692039